cmd/intervalContacts/intervalContacts.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/bed/bedpe"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/interval"
+	"log"
+)
+
+func intervalContacts(bedpeFile string, inFile string, contactOutFile string) {
+	var inIntervals = make([]interval.Interval, 0)
+	var currOverlaps []interval.Interval
+	var err error
+
+	out := fileio.EasyCreate(contactOutFile)
+
+	contactRecords := bedpe.Read(bedpeFile)
+	inChan := interval.GoReadToChan(inFile)
+
+	for i := range inChan {
+		inIntervals = append(inIntervals, i)
+	}
+	inTree := interval.BuildTree(inIntervals)
+
+	for _, i := range contactRecords {
+		currOverlaps = interval.Query(inTree, i.A, "any")
+		if len(currOverlaps) > 0 {
+			bed.WriteBed(out, i.B) // if anything in the input set overlaps i.A, write i.B to the output.
+		}
+		currOverlaps = interval.Query(inTree, i.B, "any")
+		if len(currOverlaps) > 0 {
+			bed.WriteBed(out, i.A) // and vice versa
```

Comment:
Can I ask why you chose to write the interacting (other) bed region in the bedpe as the output, and not include the overlapping bed region in the bedpe or the entry in the interval inFile in the output? Is this best format for downstream applications? I guess you can grep for lines in the output in the bedpe and interval inFile with the line number to find all 3 regions (bedpe overlapping, bedpe interacting, inFile interval entry). I am curious because I might want to use this program to find interacting regions for UHIs as well.